mittab/libs/assign_rooms.py

```diff
@@ -0,0 +1,64 @@
+import random
+from django.db import transaction
+from mittab.apps.tab.models import RoomCheckIn, Round, TabSettings
+from mittab.libs import errors, mwmatching, tab_logic
+
+
+def add_rooms():
+    no_seeding = TabSettings.get("disable_room_seeding", 0)
+
+    # Clear any existing room assignments
+    Round.objects.filter(round_number=TabSettings.get(
+        "cur_round") - 1).update(room=None)
+    round_number = TabSettings.get("cur_round") - 1
+
+    rooms = RoomCheckIn.objects.filter(
+        round_number=round_number).select_related("room")
+    rooms = sorted((r.room for r in rooms), key=lambda r: r.rank, reverse=True)
+    pairings = tab_logic.sorted_pairings(round_number)
+
+    if no_seeding:
+        random.shuffle(pairings)
+
+    if not pairings or not rooms or len(pairings) > len(rooms):
+        raise errors.RoomAssignmentError("Not enough rooms or pairings")
+
+    graph_edges = []
+
+    for pairing_i, pairing in enumerate(pairings):
+        for room_i, room in enumerate(rooms):
+            weight = 0
+
+            # High seed high room bonus
+            if no_seeding == 0:
+                weight -= abs(pairing_i - room_i)
+
+            # Bad room penalty
+            weight -= room.rank
```

Comment:
I don't fully understand the value of a bad room penalty here. We're already checking the relative mis-match with the pairing priority above

Is the idea here to account for scenarios where there are more rooms than pairings, and therefore we want an additional penalty for the worst rooms, to make sure they're de-prioritized? If so:
 1. Can we get away with just ignoring the bottom-ranked rooms?
 2. If we can't (perhaps this wouldn't work with room tags), I think we may wanna make this penalty even stronger. 1 point of room rating feels more significant than 1 level of mis-match in the pairing priority, and right now they're weighted equally. I'd suggest `-= room.rank * 100`